### PR TITLE
Typescript config

### PR DESCRIPTION
**Pull request by aczajczynska**
Typescript config, babel config, tsconfig, styled components.

### CODE REVIEW

File: dist/App.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { createAppContainer } from 'react-navigation';
+import { createStackNavigator } from 'react-navigation-stack';
+import { navigationStructure } from './app/modules';
+// import { AppWrapper } from './styled';
+import { View } from 'react-native';
+// import ThemeProvider from 'styled-components';
+// import { defaultTheme } from './app/styles/themes/defaultTheme';
+const AppNavigator = createStackNavigator(navigationStructure);
+const AppContainer = createAppContainer(AppNavigator);
+export default class App extends PureComponent {
```
Comment (lukaszblasz) [inline dist/App.js]: maybe it should be just functional component, because it is doning nothing. else than render

File: dist/App.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react';
+import { createAppContainer } from 'react-navigation';
+import { createStackNavigator } from 'react-navigation-stack';
+import { navigationStructure } from './app/modules';
+// import { AppWrapper } from './styled';
+import { View } from 'react-native';
+// import ThemeProvider from 'styled-components';
```
Comment (lukaszblasz) [inline dist/App.js]: commented code should be removed

File: dist/app/modules/MainScreen/MainScreen.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import Modules from '../index';
+class MainScreen extends PureComponent {
+    constructor() {
+        super(...arguments);
+        this.handleToSecondScreen = () => {
+            this.props.navigation.navigate(Modules.ProfileScreenModule.name);
+        };
+    }
+    render() {
+        return (React.createElement(MainWrapper, null,
```
Comment (lukaszblasz) [inline dist/app/modules/MainScreen/MainScreen.js]: Why React.createElement?
Please ise jsx sytax only and remove all createElements from it.

File: dist/App.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+// import { defaultTheme } from './app/styles/themes/defaultTheme';
+const AppNavigator = createStackNavigator(navigationStructure);
+const AppContainer = createAppContainer(AppNavigator);
+export default class App extends PureComponent {
+    render() {
+        return (React.createElement(View, { style: { flexGrow: 1 } },
+            React.createElement(AppContainer, null)));
```
Comment (lukaszblasz) [inline dist/App.js]: change all js to jsx and use standard jsx syntax in all the places.

File: dist/App.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react';
```
Comment (adamkosmala) [inline dist/App.js]: please do not commit `dist` folder

File: dist/app/modules/MainScreen/MainScreen.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+            React.createElement(MainTitle, null, "Espeo Create React Native App"),
```
Comment (lukaszblasz) [inline dist/app/modules/MainScreen/MainScreen.js]: all texts shoudl be in react intl. in separate language files. React intl will be done as a next thing ? BtW soonner the better, less places to be refactored later on.

File: jest.config.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    }
+  },
+  "moduleDirectories": ["node_modules", "src"],
+  "moduleFileExtensions": ["ts", "tsx", "html", "js", "json"],
+  "moduleNameMapper": {
+    "^@core/(.*)": "<rootDir>/src/app/$1",
+    "^@assets/(.*)": "<rootDir>/src/assets/$1",
+    "^@environments/(.*)": "<rootDir>/src/environments/$1"
+  },
+  "setupFiles": ["<rootDir>/test-setup.js"]
```
Comment (adamkosmala) [inline jest.config.json]: please take the latest `jest.config.json` from CRA repository- it has updated aliases and config

File: src/app/modules/MainScreen/MainScreen.tsx [hunk 8]
```diff
@@ -1,9 +1,13 @@
 import React, { PureComponent } from 'react';
-import { MainWrapper, MainTitle } from './MainScreen.styles';
+import { MainWrapper, MainTitle } from './styles';
 import { Button } from 'react-native';
 import Modules from '../index';
 
-class MainScreen extends PureComponent {
+interface IProps {
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/MainScreen.tsx]: we do not use `I` prefixes anymore

File: dist/app/modules/MainScreen/styles.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+// import styled from 'styled-components';
+import { styled } from '../../styles/themes/defaultTheme';
+export const MainWrapper = styled.View `
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+`;
+export const MainTitle = styled.Text `
```
Comment (lukaszblasz) [inline dist/app/modules/MainScreen/styles.js]: Please create some generic Text component which will have some general styling like font-family etc. and specific components inside modules should inherit from it.

File: src/app/modules/ProfileScreen/ProfileScreen.tsx [hunk 10]
```diff
@@ -1,9 +1,12 @@
 import React, { PureComponent } from 'react';
 import { Button } from 'react-native';
-import { ProfileWrapper, ProfileTitle } from './ProfileScreen.styles';
-import Modules from '../index';
+import { ProfileWrapper, ProfileTitle } from './styles';
 
-class ProfileScreen extends PureComponent {
+interface IProps {
```
Comment (adamkosmala) [inline src/app/modules/ProfileScreen/ProfileScreen.tsx]: prefix

File: dist/app/modules/MainScreen/styles.js (hunk 9 above)
Comment (lukaszblasz) [reply dist/app/modules/MainScreen/styles.js]: something like 

export const MainTitle = styled(MyGenericText)

File: dist/modules/MainScreen/MainScreen.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+class MainScreen extends PureComponent {
+    constructor() {
+        super(...arguments);
+        this.handleToSecondScreen = () => {
+            this.props.navigation.navigate(Modules.ProfileScreenModule.name);
+        };
+    }
+    render() {
+        return (React.createElement(MainWrapper, null,
+            React.createElement(MainTitle, null, "Espeo Create React Native App"),
```
Comment (lukaszblasz) [inline dist/modules/MainScreen/MainScreen.js]: JSX please :)

File: src/app/modules/MainScreen/styles.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import { styled } from '../../styles/themes/defaultTheme';
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/styles.ts]:
```
import { styled } from '@core/styles/themes/defaultTheme';
```
?

File: dist/modules/MainScreen/styles.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components';
+export const MainWrapper = styled.View `
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+`;
+export const MainTitle = styled.Text `
```
Comment (lukaszblasz) [inline dist/modules/MainScreen/styles.js]: same here a lot of duplicated styles, there are few styled component which is doing the same.

File: tsconfig.json [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "jsx": "react-native",
+    "jsxFactory": "React.createElement",
+    "baseUrl": "./src",
+    "paths": {
+      "@core/*": ["app/*"],
+      "@modules/*": ["modules/*"],
+      "@assets/*": ["assets/*"],
+      "@styles/*": ["app/styles/*"]
```
Comment (adamkosmala) [inline tsconfig.json]: do we have all aliases there? please make sure that this file, webpack config, jest config and eslint config includes the same paths' aliases

File: src/app/modules/MainScreen/MainScreen.tsx [hunk 15]
```diff
@@ -1,9 +1,13 @@
...
+  navigation: any;
```
Comment (lukaszblasz) [inline src/app/modules/MainScreen/MainScreen.tsx]: Probably navigation props types can be imported. Please remove any.
https://reactnavigation.org/docs/en/typescript.html

**Comment by adamkosmala** [comment]
Name of this PR and branch is misleading- what exactly are you trying to add to the existing codebase? please provide more meaningful PR name and some sort of description (with bullet points, for example)

File: src/app/modules/MainScreen/styles.ts (hunk 12 above)
Comment (lukaszblasz) [inline src/app/modules/MainScreen/styles.ts]: dont use relative imports. Please use something like @styles.themes. To achive that there is a need for babel with proper plugin for aliases,.

File: tsconfig.json [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+    "strictFunctionTypes": true,
+    "strictNullChecks": true,
+    "strictBindCallApply": true,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "jsx": "react-native",
+    "jsxFactory": "React.createElement",
+    "baseUrl": "./src",
+    "paths": {
+      "@core/*": ["app/*"],
```
Comment (lukaszblasz) [inline tsconfig.json]: raect native not use paths from tsconfig BTW. They can be removed becasue they are not used for anything.

File: tsconfig.json (hunk 14 above)
Comment (lukaszblasz) [reply tsconfig.json]: Aliases not working from tsconfig in react native :) they need to be in babel config :)

Comment (lukaszblasz) [reply tsconfig.json]: Should be removed form there to not confuse.

File: dist/app/modules/MainScreen/MainScreen.js (hunk 6 above)
Comment (lukaszblasz) [reply dist/app/modules/MainScreen/MainScreen.js]: I see its dist catalog :) Please add it to gitignore and remove :)

File: dist/App.js (hunk 4 above)
Comment (lukaszblasz) [reply dist/App.js]: remove dist and everything inside :) and add proper gitignore :)

File: dist/App.js (hunk 5 above)
Comment (lukaszblasz) [reply dist/App.js]: +1

File: jest.config.json (hunk 7 above)
Comment (lukaszblasz) [reply jest.config.json]: Not user if jest will work with react native out fo the box... with teh same config, but lets left unit tests fo native as a next thing not part of this story.

File: dist/app/modules/MainScreen/MainScreen.js (hunk 3 above)
Comment (lukaszblasz) [reply dist/app/modules/MainScreen/MainScreen.js]: just remove dist :)

File: tsconfig.json [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    "noUnusedParameters": true,
+    "resolveJsonModule": true,
+    "strictFunctionTypes": true,
+    "strictNullChecks": true,
+    "strictBindCallApply": true,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "jsx": "react-native",
+    "jsxFactory": "React.createElement",
+    "baseUrl": "./src",
```
[the remark this answers is not in the document]
Comment (aczajczynska) [reply tsconfig.json]: baseUrl?

[the remark this answers is not in the document]
Comment (adamkosmala) [reply tsconfig.json]: yap

**Pull request merged**